### ISSUE TITLE
[huawei_analytics] Issue found: Violation of User Data policy

### ISSUE THREAD

**Issue by darkang3lz92** [HMS-Core/hms-flutter-plugin#395]
#### Description

Recently received user data policy violation as below:

<details>
<summary> Issue found: Violation of User Data policy</summary>

We reviewed SDKs used by your app and found non-compliant version(s) of SDK(s) which collects persistent device identifiers. Persistent device identifiers may not be linked to other personal and sensitive user data or resettable device identifiers.

We have observed that your app is using the Huawei Hianalytics.
Your app may face additional enforcement actions, if you do not resolve this issue by November 13, 2024.
Issue details

We found an issue in the following area(s):

SDK: Huawei HMA Hianalytics SDK com.huawei.hms:hianalytics
To bring your app into compliance, follow these steps:

Huawei HMA Hianalytics SDK com.huawei.hms:hianalytics: Consider upgrading to a policy-compliant version of the SDK if available from your SDK provider, or removing the SDK.
</details>

It's seems happen to the HiAnalytics library.

#### Library version
huawei_analytics: 6.12.0+301

**Comment by ahmeddhus** [comment]
Any updates?

**Comment by mahmuttaskiran** [comment]
We encountered the same issue and removed the Huawei Analytics SDK for Google Play only.

**Comment by ahmeddhus** [comment]
@mahmuttaskiran We did the same, but did the error on Google Play get resolved after publishing the version without the Huawei Analytics SDK?

**Comment by mahmuttaskiran** [comment]
> @mahmuttaskiran We did the same, but did the error on Google Play get resolved after publishing the version without the Huawei Analytics SDK?

Yes, but we needed to delete the app bundles rejected by this issue before uploading a new version for some reason.

**Comment by ahmeddhus** [comment]
> > @mahmuttaskiran We did the same, but did the error on Google Play get resolved after publishing the version without the Huawei Analytics SDK?
> 
> Yes, but we needed to delete the app bundles rejected by this issue before uploading a new version for some reason.

We uploaded the beta version without the SDK, but the issue still persists. From what I understand, we need to publish the new version first. Once that's done, we should delete the app bundles that were previously rejected. After publishing the beta version, this should resolve the issue. Is that correct?

**Comment by darkang3lz92** [comment]
> > > @mahmuttaskiran We did the same, but did the error on Google Play get resolved after publishing the version without the Huawei Analytics SDK?
> > 
> > 
> > Yes, but we needed to delete the app bundles rejected by this issue before uploading a new version for some reason.
> 
> We uploaded the beta version without the SDK, but the issue still persists. From what I understand, we need to publish the new version first. Once that's done, we should delete the app bundles that were previously rejected. After publishing the beta version, this should resolve the issue. Is that correct?

Generally you will need to unpublished affected bundle/apk in all release channel (internal, alpha, beta, production) if any. You will need to upload the new version into each channels build that contain HiAnalytics library as Google didn't provide any ways to remove build without uploading a new build to replace it.
